src/main/java/com/celements/webform/DocFormScriptService.java

```diff
@@ -37,29 +37,33 @@ private XWikiContext getContext() {
   /*
    * TODO: Please get rid of throwing an exception to the view (client), use try/catch
    * and write the exception in a log-file
+   * 
+   * ATTENTION: use only for preview and NOT for save! Removed objects will not be saved
+   * correctly using this method. To save use updateAndSaveDocFromRequest() instead.
    */
   public Set<Document> updateDocFromMap(DocumentReference docRef, Map<String, ?> map
       ) throws XWikiException {
-    Map<String, String[]> recompMap = new HashMap<String, String[]>();
-    for (String key : map.keySet()) {
-      if(map.get(key) instanceof String[]) {
-        recompMap.put(key, (String[])map.get(key));
-      } else if(map.get(key) instanceof String) {
-        recompMap.put(key, new String[]{(String)map.get(key)});
-      }
-    }
-    if (LOGGER.isTraceEnabled()) {
-      LOGGER.trace("recompiled map '{}' to '{}'", map, recompMap);
-    }
-    Set<Document> docs = new HashSet<Document>();
     Collection<XWikiDocument> xdocs = getDocFormCommand().updateDocFromMap(docRef,
-        recompMap, getContext());
+        getDocFormCommand().prepareMapForDocUpdate(map), getContext());
+    Set<Document> docs = new HashSet<Document>();
     for (XWikiDocument xdoc : xdocs) {
       docs.add(xdoc.newDocument(getContext()));
     }
     return docs;
   }
   
+  public Map<String, Set<DocumentReference>> updateAndSaveDocFromMap(
+      DocumentReference docRef, Map<String, ?> map) {
+    Collection<XWikiDocument> xdocs = Collections.<XWikiDocument>emptyList();
+    try {
+      xdocs = getDocFormCommand().updateDocFromMap(docRef, getDocFormCommand(
+          ).prepareMapForDocUpdate(map), getContext());
+    } catch (XWikiException xwe) {
+      LOGGER.error("Exception in getDocFormCommand().updateDocFromMap()", xwe);
+    }
+    return getDocFormCommand().saveXWikiDocCollection(xdocs);
```

Comment:
Please check for edit-access rights before saving the xdocs. I suggest to save all or none (transaction behaviour).